test/sim/moves/shelltrap.js

```diff
@@ -32,4 +32,13 @@ describe('Shell Trap', function () {
 		battle.makeChoices('move shelltrap, move splash', 'move tackle, move splash');
 		assert.equal(move.pp, move.maxpp - 2);
 	});
+
+	it('should not power up if hit by a Z- or Max move', function () {
```

Comment:
```suggestion
	it('should not Max if hit by a Max move', function () {
```

Reply:
The bug also affects Z-moves, I was just too lazy to include both tests...

Reply:
I know, I just think clarifying this is more important than mentioning Z moves.

You can mention Z-moves in a comment if you really care? But I don't think it's necessary.

Reply:
... I've added the other test now, so that should make it clearer.